tests/platform_tests/test_cpu_memory_usage.py

```diff
@@ -35,36 +40,191 @@ def test_cpu_memory_usage(duthosts, enum_rand_one_per_hwsku_hostname, setup_thre
+    cpu_usage_program_to_check = []
+
+    prepare_ram_cpu_usage_results(MonitResult, counterpoll_cpu_usage_threshold, memory_threshold, monit_results, outstanding_mem_polls,
+                                  outstanding_procs, outstanding_procs_counter, program_to_check,
+                                  cpu_usage_program_to_check)
+
+    log_cpu_usage_by_vendor(cpu_usage_program_to_check, counterpoll_type)
+
+    cpu_usage_average = caculate_cpu_usge_average_value(extract_valid_cpu_usage_data(cpu_usage_program_to_check, poll_interval), cpu_usage_program_to_check)
+    logging.info("Average cpu_usage is {}".format(cpu_usage_average))
+    assert cpu_usage_average < counterpoll_cpu_usage_threshold, "cpu_usage_average of {} exceeds the cpu threshold:{}".format(program_to_check, counterpoll_cpu_usage_threshold)
+    assert not outstanding_mem_polls, " Memory exceeds the memory threshold {} ".format(outstanding_mem_polls, memory_threshold)
+
+
+def log_cpu_usage_by_vendor(cpu_usage_program_to_check, counterpoll_type):
+    if cpu_usage_program_to_check:
+        logging.info('CPU usage for counterpoll type {} : {}'.format( counterpoll_type, cpu_usage_program_to_check))
+
+
+def get_manufacturer_program_to_check(duthost):
+    if is_mlnx(duthost):
+        return CounterpollConstants.SX_SDK
+
+
+def is_mlnx(duthost):
+    if CounterpollConstants.MLNX_PLATFORM_STR in duthost.facts["platform"]:
+        return True
+
+
+def prepare_ram_cpu_usage_results(MonitResult, cpu_threshold, memory_threshold, monit_results, outstanding_mem_polls,
+                                  outstanding_procs, outstanding_procs_counter, program_to_check,
+                                  program_to_check_cpu_usage):
+    for i, monit_result in enumerate(MonitResult(*_) for _ in monit_results):
+        logging.debug("------ Iteration %d ------", i)
+        check_memory(i, memory_threshold, monit_result, outstanding_mem_polls)
+        for proc in monit_result.processes:
+            update_cpu_usage_desired_program(proc, program_to_check, program_to_check_cpu_usage)
+
+
+def extract_valid_cpu_usage_data(program_to_check_cpu_usage, poll_interval):
+    """
+    This method it to extract the valid cpu usage data according to the poll_interval
+    1. Find the index for the max one for every poll interval,
+    2. Discard the data if the index is on the edge(0 o the length of program_to_check_cpu_usage -1)
+    3. If the index is closed in the neighbour interval, only keep the former one
+    4. Return all indexes
+    For example:
+    poll_interval = 10
+    7, 1, 0, 1, 0, 1, 5, 1, 1,2, 0, 1, 0, 1, 0, 6, 1, 1, 1,2
+    return [15]
+    0, 1, 0, 1, 0, 1, 0, 1, 0, 8, 7, 1, 0, 1, 0, 6, 1, 1, 1,2
+    return [9]
+    """
+    valid_cpu_usage_center_index_list = []
+    poll_number = len(program_to_check_cpu_usage) // poll_interval
+
+    def find_max_cpu_usage(cpu_usage_list, poll_times):
+        max_cpu_usage = cpu_usage_list[0]
```

Comment:
to get max cpu usage from a list, you can use max(cpu_usage_list), and we don't need index here. If you don't want to count begin and end, just do it like max(cpu_usage_list[1:-1])